src/prefect/cli/execute.py

```diff
@@ -62,13 +63,19 @@ def cloud_flow():
         storage_schema = prefect.serialization.storage.StorageSchema()
         storage = storage_schema.load(flow_data.storage)
 
-        flow = storage.get_flow(storage.flows[flow_data.name])
-        environment = flow.environment
+        ## populate global secrets
+        secrets = prefect.context.get("secrets", {})
+        for secret in storage.secrets:
+            secrets[secret] = PrefectSecret(name=secret).run()
 
-        environment.setup(storage=storage)
-        environment.execute(
-            storage=storage, flow_location=storage.flows[flow_data.name]
-        )
+        with prefect.context(secrets=secrets):
```

Comment:
We may need to propagate secrets down during execution. I don't think this will work for flows using environments like the `DaskKubernetesEnvironment` because the `execute` step creates a "scheduler" job that calls the [run_flow](https://github.com/PrefectHQ/prefect/blob/master/src/prefect/environments/execution/dask/k8s.py#L261) function from the environment

Reply:
Oh nevermind it looks like you already do this!